bertserini/reader/base.py

```diff
@@ -0,0 +1,114 @@
+from typing import List, Union, Optional, Mapping, Any
+import abc
+
+__all__ = ['Question', 'Context', 'Reader', 'Answer', 'TextType']
+
+
+TextType = Union['Question', 'Context', 'Answer']
+
+
+class Question:
+    """
+    Class representing a question.
+    A question contains the question text itself and potentially other metadata.
+    Parameters
+    ----------
+    text : str
+        The question text.
+    id : Optional[str]
+        The question id.
+    """
+    def __init__(self, text: str, id: Optional[str] = None, language: str = "en"):
+        self.text = text
+        self.id = id
+        self.language = language
+
+
+class Context:
+    """
+    Class representing a Context to find answer from.
+    A text is unspecified with respect to it length; in principle, it
+    could be a full-length document, a paragraph-sized passage, or
+    even a short phrase.
+    Parameters
+    ----------
+    text : str
+        The context that contains potential answer.
+    metadata : Mapping[str, Any]
+        Additional metadata and other annotations.
+    score : Optional[float]
+        The score of the context. For example, the score might be the BM25 score
+        from an initial retrieval stage.
+    """
+
+    def __init__(self,
+                 text: str,
+                 language: str = "en",
+                 metadata: Mapping[str, Any] = None,
+                 score: Optional[float] = 0):
+        self.text = text
+        self.language = language
+        if metadata is None:
```

Comment:
metadata is from parameters, this block is to assign the self.metadata an empty dict if metadata is None. If it is not None, it will still assign what it should be.